src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -681,10 +711,36 @@ private static bool IsArrayCompatibleReadOnlyInterface(Type type)
             return allProperties;
         }
 
+        [RequiresUnreferencedCode(PropertyTrimmingWarningMessage)]
+        private static object? GetParameterValue(ParameterInfo property, IConfiguration config, BinderOptions options)
+        {
+            string parameterName = property.Name!;
+
+            var propertyBindingPoint = new BindingPoint(
+                initialValueProvider: () => config.GetSection(parameterName).Value,
```

Comment:
This doesn't seem right. `initialValueProvider` is used when the default value for the `BindingPoint` can be lazily retrieved.

In the case of binding a parameter, the default value should just be the parameter's default value (if it has one).